### ISSUE TITLE
libexpat 2.2.1 and later no longer supports VC++ 6 or visual studio 2008 ?

### ISSUE THREAD

**Issue by billysuh** [libexpat/libexpat#112]
With 2.2.0 I was able to compile it with visual studio 2008 by taking the DSW file for VC++ 6.0 and converting it for visual studio 2008.

With 2.2.1 and later I can't seem to find any way to use vcbuild.exe to compile libexpat. I know it is old, but our product still needs to use vcbuild. 

Is there any recourse for this situation ? Otherwise we'd need to take up a new version of VC and re-compile all libs we use, not just libexpat. Thanks!

**Comment by hartwork** [comment]
Hi @billysuh

we still have project files for Visual Studio 200x around:
```console
# find -name \*.vcxproj | sort
./examples/elements.vcxproj
./examples/outline.vcxproj
./lib/expat_static.vcxproj
./lib/expat.vcxproj
./lib/expatw_static.vcxproj
./lib/expatw.vcxproj
./tests/benchmark/benchmark.vcxproj
./tests/runtests.vcxproj
./xmlwf/xmlwf.vcxproj
```
These don't work for you?

**Comment by billysuh** [comment]
@hartwork thank you for your reply.

Unfortunately no. I tried to load ./lib/expat_static.vcxproj in Visual Studio 2008, which popped up a window for conversion (saying the project was created in a previous version of Visual Studio which isn't probably true). The conversion then fails saying it fails to load the file. This is unlike libexpat 2.2.0 where I could load expat.vcproj without any issue (and no conversion was needed). 

AFAIK these vcxproj files are meant for Visual Studio 2010 and they seem to have gone through an upgrade because libexpat 2.2.0 came with vcproj (vs vcxproj). Also, libexpat 2.2.0's dsw and dsp files worked just as well with visual studio 2008, but these files are gone beginning with 2.2.1. I happen to want to compile libexpat 2.2.3 because that has the latest security fixes.

If you could help me regarding this in any way, that'd be great... 

Best regards
Billy

**Comment by hartwork** [comment]
Hi!

Judging from 309ad17f97fe1e59a000d69a6fb1c99c76a6fe6b, our current project files seem to be for Visual Studio 2013 and later. We dropped the VC++ 6 ones in a67f2a2545ca2136557af2f8d2c08e87d297b47a so direct predecessor c808687f32f47396079b62c1c5690646ab7e22ac is the last commit with them, a [zip download](https://github.com/libexpat/libexpat/archive/c808687f32f47396079b62c1c5690646ab7e22ac.zip) of that is available. To use these files with 2.2.3 code, you would need to add `lib/siphash.h` and `lib/loadlibrary.c` to these at the right places. I would expect the result to compile with Visual Studio 2008, since c74d380b8f0b5215153d840d2af9455d9c53b960 is contained in 2.2.3.

**Comment by dr0biwan** [comment]
Just wanted to add that 2.2.3 does compile with VS2008 when using the existing 2.2 project files after adding in the new files (is how I've been updating my working copy)  The only issue I remember was needing to include stdlib.h in one of the newer files (will have to check that later) otherwise it works.

-dro

**Comment by kwaclaw** [comment]
I believe one should be able to build Expat using the free Visual Studio
Community edition.

**Comment by hartwork** [comment]
@dr0biwan, I'm happy to include `stdlib.h` where needed if there is a bug report or a pull request.

**Comment by billysuh** [comment]
@hartwork Thank you for your original suggestion, which I followed by mentioning those .h and .c files in the project file to get the compilation to work.

For what it's worth I also tried to see what @dr0biwan meant regarding stdlib.h. I simply took (expat_static.dsp - the only project I need) and dropped into 2.2.3 code under lib/ and ran the build. It actually went OK, with this warning:

```
1>c:\users\administrator\downloads\expat 2.2.3\source\lib\xmlparse.c(907) : warning C4996: 'getenv': This function or variable may be unsafe. Consider using _dupenv_s instead. To disable deprecation, use _CRT_SECURE_NO_WARNINGS. See online help for details.
1>        c:\program files\microsoft visual studio 9.0\vc\include\stdlib.h(447) : see declaration of 'getenv'
```
Indeed, xmlparse.c (907) included stdlib.h from my VS2008 to use getenv which is supposedly not safe? Can I still just ignore the warning or is there really a need to do something ? (Again I am only talking about expat_statlic.dsp).
